docs/ChainOrderTransitivity.md

```diff
@@ -0,0 +1,153 @@
+The resulting is transitive, hence total.
+
+The motivation for the opcert number tiebreaker is described in the ["Design Specification for Delegation and Incentives in Cardano"][ledger-readme].
+We describe our understanding of this use case for the opcert number tiebreaker:
+
+Consider an attacker who corrupts an SPO's node with a specific issuer identity, including the hot key, but not the cold key.
+They can now forge blocks on behalf of this issuer identity, deciding which transactions are included in the block, potentially spoiling the reputation of the SPO in the process.
+(Rewards still go to the account of the issuer identity, which the attacker can not access.)
+The attacked SPO can now issue a new hot key with a higher opcert number, and start a new node with it installed.
+Now, in every slot where this issuer identity is elected, both the attacker and the attacked SPO can forge a block.
+
+ - If the SPO forges a strictly longer or shorter chain, the opcert tiebreaker does not matter, citing from the aforementioned specification:
+   > Note that nodes take the precedence amongst operational key certificates into account only after comparing the length of the chains.
+   > When the node is already up to date and receives two conflicting blocks that add to its current chain, the length will of course always be the same.
+   > But this rule is important: if we did not compare the lengths of the chains before giving preference to the block with the newer operational certificate, it would be possible to force a node to do a rollback of arbitrary length, by sending it a block from a past slot, signed using a newer certificate than the block that the node already has in its chain for that slot.
+   > This would open up an attack where a stake pool operator could force nodes to do arbitrary rollbacks.
+ - If the SPO forges on top of a chain that is just as long as the one the attacker forged on, the opcert number tiebreaker gives precedence to the SPO as they have the higher opcert number.
+   Note that this is the usual case when two nodes are elected in the same slot.
+
+An important observation is that the VRF is the same for both blocks, as the VRF does not depend on the hot key.
+This means _swapping_ the order of the opcert number and the VRF tiebreaker also satisfies the requirements of the opcert number mechanism as just described (again, this _was_ the original implementation before [ouroboros-network#2348](https://github.com/IntersectMBO/ouroboros-network/pull/2348)).
+
+Note that there is a difference: With the current chain order, a block `B` with a higher opcert number will be preferred over an attacker's block `B'` with the same block number even if it is in a different slot, as opposed to a random tiebreak using the VRF when using the proposed order with these two tiebreakers swapped to restore transitivity.
```

Comment:
"even if it is in a different slot +(despite being the same length)+"

Reply:
Wouldn't this lead to stating the same thing (same block number/length almost immediately next to each other)?

> with the same block number even if it is in a different slot despite being the same length

Reply:
Oh dear. I had missed the "with the same block number" when I was reading.